wp-content/civi-extensions/civiglific/Civi/MonetaryReceiptService.php

```
@@ -0,0 +1,225 @@
+      $saveDir = $uploadBase . '/uploads/civicrm/persist/contribute/contribution/';
+
+      if (!file_exists($saveDir)) {
+        mkdir($saveDir, 0755, TRUE);
+      }
+
+      $savePath = $saveDir . $fileNameForPdf;
+      if (@file_put_contents($savePath, $pdfBinary) === FALSE) {
+        throw new \Exception("Failed to save PDF to {$savePath}");
+      }
+
+      $baseUrl = rtrim(\CRM_Core_Config::singleton()->userFrameworkBaseURL, '/');
```

PHPMD (2.15.0)

Avoid unused local variables such as '$baseUrl'. (undefined)

(UnusedLocalVariable)